tests/integration/test_pip_packages.py

```diff
@@ -24,3 +28,184 @@ def test_failing_pip_local_path(test_env):
+        utils.assert_expected_files(source_path, expected_files, tmpdir)
+
+        purl = env_data["purl"]
+        deps_purls = []
+        source_purls = []
+        if "dep_purls" in env_data:
+            deps_purls = [{"purl": x} for x in env_data["dep_purls"]]
+        if "source_purls" in env_data:
+            source_purls = [{"purl": x} for x in env_data["source_purls"]]
+
+        image_contents = [{"dependencies": deps_purls, "purl": purl, "sources": source_purls}]
+        utils.assert_content_manifest(client, completed_response.id, image_contents)
+
+    def replace_hashes(orig_str):
+        """
+        Replace expected commits and hashes.
+
+        :param orig_str: original string
+        :return: string with replaced values
+        :rtype: str
+        """
+        replace_rules = {
+            "FIRST_DEP_COMMIT": new_dep_commits[0],
+            "SECOND_DEP_COMMIT": new_dep_commits[1],
+            "FIRST_DEP_HASH": dep_hash,
+            "MAIN_REPO_COMMIT": main_repo_commit,
+        }
+        res_string = copy(orig_str)
+        for s, r in replace_rules.items():
+            if s in res_string:
+                res_string = res_string.replace(s, r)
+        return res_string
+
+    def update_expected_data():
+        """
+        Update expected data for the test in place.
+
+        Change commits and hashes in:
+        * expected_files
+        * response_expectations
+        * all purls in env_data
+        """
+        new_expected_files = {}
+        for file, url in env_data["expected_files"].items():
+            new_expected_files[replace_hashes(file)] = replace_hashes(url)
+        env_data["expected_files"] = new_expected_files
+
+        for i, dep in enumerate(env_data["response_expectations"]["dependencies"]):
+            env_data["response_expectations"]["dependencies"][i]["version"] = replace_hashes(
+                dep["version"]
+            )
+        for i, dep in enumerate(env_data["response_expectations"]["packages"][0]["dependencies"]):
+            env_data["response_expectations"]["packages"][0]["dependencies"][i][
+                "version"
+            ] = replace_hashes(dep["version"])
+
+        env_data["purl"] = replace_hashes(env_data["purl"])
+        for i, p in enumerate(env_data["dep_purls"]):
+            env_data["dep_purls"][i] = replace_hashes(p)
+        for i, p in enumerate(env_data["source_purls"]):
+            env_data["source_purls"][i] = replace_hashes(p)
+
+    env_data = utils.load_test_data("pip_packages.yaml")["cached_deps"]
+    # Download dependency repo into a new directory
+    dep_repo_dir = os.path.join(tmpdir, "dep")
+    os.mkdir(dep_repo_dir)
+    cloned_dep_repo = git.Repo.clone_from(env_data["ssh_git_dep"], dep_repo_dir)
+    # Open a new branch in dependency repo
+    cloned_dep_repo.git.checkout("-b", env_data["test_branch"])
+    assert cloned_dep_repo.active_branch.name == env_data["test_branch"]
+
+    try:
```

Comment:
Do we really need to have everything in this try block? In other words, what sort of error are we expecting to happen here and which calls could raise them

Reply:
@athos-ribeiro I moved `try` for all Cachito requests and main test assertion. Even with a broken test we expect to delete the branch.

Reply:
Could you use setup/teardown functions to do that for you instead?

Reply:
Ok! Done. The setup method is almost useless :D But thank you for help with teardown method.